.github/workflows/test.yml

```diff
@@ -10,20 +10,43 @@ on:
 jobs:
   test:
     name: Unit test
-    runs-on: [ubuntu-latest]
+    runs-on: [self-hosted]
+    env:
+      CARGO_INCREMENTAL: 0
+      SKIP_BUILD_WASM: true
+      BUILD_DUMMY_WASM_BINARY: true
+      LLVM_PROFILE_FILE: "llvmcoveragedata-%p-%m.profraw"
+      WASM_TARGET_DIRECTORY: /tmp/wasm
+      RUSTFLAGS: "-Zinstrument-coverage"
     steps:
       - name: Checkout Repository
         uses: actions/checkout@master
 
-      - name: Install Rust toolchain ${{ matrix.rust }}
+      - name: Install Rust toolchain
         uses: actions-rs/toolchain@v1
         with:
-          components: rustfmt, clippy
+          toolchain: nightly
+          components: rustfmt, clippy, llvm-tools-preview
           target: wasm32-unknown-unknown
           override: true
 
       - name: cache dependencies
         uses: Swatinem/rust-cache@v1
-      
-      - name: Test
-        run: cargo test -- --nocapture --quiet
+
+      - name: show rustc version
+        run: rustc --version
+
+      - name: test
+        run: cargo test --workspace --exclude node-executor --exclude node-cli
```

Comment:
why exclude ?

Reply:
node-executor contains some wasm tests that llvm coverage tool doesn't support.
node-cli contains need to test kill process, some tests fail on github actions. see: https://github.com/deeper-chain/deeper-chain/runs/4664561819?check_suite_focus=true
![image](https://user-images.githubusercontent.com/4054836/147739348-774fd0e3-db64-4218-8488-f3941dc649be.png)

Reply:
the unit test in dev branch succced;
waht cause this branch not pass the test

Reply:
I don't know why, maybe llvm source code coverage doesn't work on wasm code. I'm following test instructions from https://github.com/paritytech/polkadot/blob/83e5955b83870c0b107af37934d9f8a4ecf2c63d/doc/testing.md
if I use `cargo test` I can't compile `node-executor`, If using `cargo test --exclude node-executor` everything goes fine.